model.py

```diff
@@ -52,12 +52,15 @@ def __init__(self, config, word_dict=None, classes=None, ckpt=None):
         self.config.num_classes = len(self.classes)
 
         embed_vecs = self.word_dict.vectors
-        self.network = get_network(config, embed_vecs).to(self.device)
+        self.network = getattr(networks, config.model_name)(config, embed_vecs).to(self.device)
         self.init_optimizer()
 
         if ckpt:
             self.network.load_state_dict(ckpt['state_dict'])
             self.optimizer.load_state_dict(ckpt['optimizer'])
+        else:
+            init_weights = networks.get_weight_init_func(config)
+            self.network.apply(init_weights)
```

Comment:
They said "Pass an initialization function to `torch.nn.Module.apply`. It will initialize the weights in the entire `nn.Module` recursively."
https://stackoverflow.com/questions/49433936/how-to-initialize-weights-in-pytorch

Is that the reason the linear layers start from different weights?

Reply:
I have the same concern.. Are we able to get same results for mimic-50 as before?

Reply:
Yes, we should set `weight_init: xavier_uniform` in `caml.yml` and remove `xavier_uniform_()` in `caml.py`.
However, since the order of initialization is changed, we may not able to get the same results for MIMIC-50 as before.
It is possible to get the same results but the code would be pretty ugly...

Reply:
Solved.